### PR TITLE
Feature/signalR

### PR DESCRIPTION
**Pull request by stevenhillcox**

### CODE REVIEW

File: VotingApplication/VotingApplication.Web/Api/SignalR/ClientSignaller.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿using Microsoft.AspNet.SignalR;
+
+namespace VotingApplication.Web.Api.SignalR
+{
+    public static class ClientSignaller
+    {
+        private static IHubContext getHubContext()
```
Comment (JDawes-ScottLogic) [inline VotingApplication/VotingApplication.Web/Api/SignalR/ClientSignaller.cs]: In general private methods should be below the methods that call them, and public methods should be first.

File: VotingApplication/VotingApplication.Web/Api/SignalR/SignalHub.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.AspNet.SignalR;
+using Microsoft.AspNet.SignalR.Hubs;
+namespace VotingApplication.Web.Api.SignalR
```
Comment (JDawes-ScottLogic) [inline VotingApplication/VotingApplication.Web/Api/SignalR/SignalHub.cs]: Should be a space between `using`s and `namespace`.

File: VotingApplication/VotingApplication.Web/Scripts/Services/SignalService.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            hubOptions.listeners = {
+                'update': function () {
+                    if (updateFunction) {
+                        updateFunction();
+                    }
+                }
+            };
+
+            hubOptions.methods = ['registerObserver'];
+            hubOptions.errorHandler = function (error) {
```
Comment (JDawes-ScottLogic) [inline VotingApplication/VotingApplication.Web/Scripts/Services/SignalService.js]: Might be nicer to handle this through the `ErrorService` at some point.

**Comment by JDawes-ScottLogic** [comment]
:+1: 
Minor layout comments, happy for it to be merged.

File: VotingApplication/VotingApplication.Web/Scripts/Services/SignalService.js (hunk 3 above)
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web/Scripts/Services/SignalService.js]: Agreed

**Comment by JDawes-ScottLogic** [comment]
:+1: :shipit:

**Pull request merged**